src/core/ApolloClient.ts

```diff
@@ -124,19 +138,56 @@ export interface ApolloClientOptions<TCacheShape> {
    * @defaultValue `false`
    */
   assumeImmutableResults?: boolean;
+
+  /**
+   * @deprecated `resolvers` has been changed in Apollo Client 4.0 and will no
+   * longer be provided as an option directly to `ApolloClient`. `resolvers`
+   * should continue to be used until upgrading to 4.0.
```

Comment:
Suggestions welcome on wording here. I don't know how best to communicate "this will change, but only works as `resolvers` until 4.0"

Reply:
```suggestion
   * @deprecated in preparation to the Apollo Client 4.0: 
   * `resolvers` will change in Apollo Client 4.0 and will no longer be provided as an option directly to `ApolloClient`.
   * It is safe to keep using this option with Apollo Client 3.x.
```

Maybe something like this?